expression/builtin.go

```diff
@@ -36,6 +36,10 @@ type baseBuiltinFunc struct {
 	ctx    sessionctx.Context
 	tp     *types.FieldType
 	pbCode tipb.ScalarFuncSig
+
+	// fields used to support converting between row-based evaluation and vectorized evaluation
+	self      builtinFunc // the actual builtinFunc which inherit this baseBuiltinFunc
```

Comment:
OOP in golang looks a little wried

Reply:
The way to implement this converting is changed, PTAL.